ssl/quic/quic_impl.c

```diff
@@ -1568,19 +1568,20 @@ SSL *ossl_quic_conn_stream_new(SSL *s, uint64_t flags)
 int ossl_quic_get_error(const SSL *s, int i)
 {
     QCTX ctx;
-    int net_error;
+    int net_error, qc_last_error;
 
     if (!expect_quic(s, &ctx))
         return 0;
 
     quic_lock(ctx.qc);
     net_error = ossl_quic_channel_net_error(ctx.qc->ch);
+    qc_last_error = ctx.qc->last_error;
     quic_unlock(ctx.qc);
 
     if (net_error)
         return SSL_ERROR_SYSCALL;
 
-    return ctx.is_stream ? ctx.xso->last_error : ctx.qc->last_error;
+    return ctx.is_stream ? ctx.xso->last_error : qc_last_error;
```

Comment:
The XSO too. I would just hoist this whole ternary clause to above the `quic_unlock` call.

Reply:
@hlandau Can the xso->last_error be legally changed in concurrent threads? It would require simultaneous calls on the SSL object representing the stream from concurrent threads which, I thought, is not allowed.

Reply:
I mean, this is true, at least currently... but it doesn't really hurt to be consistent about it?